### PR TITLE
GitOps environment UI page validation

### PR DESCRIPTION
**Pull request by amitkrout**
Jira reference - https://issues.redhat.com/browse/ODC-5413

### CODE REVIEW

**Comment by amitkrout** [comment]
@gajanan-more @sanketpathak Would please let me know what i am missing 
```
$ yarn run gherkin-lint
yarn run v1.22.10
$ /Users/amit/go/src/github.com/openshift/console/frontend/packages/gitops-plugin/integration-tests/node_modules/.bin/gherkin-lint
Could not find default config file ".gherkin-lintrc" in the working directory.
To use a custom name/path provide the config file using the "-c" arg.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

File: frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature [hunk 1]
```diff
@@ -16,7 +19,10 @@ Feature: Applications managed by GitOps and Create secret
 
         @regression @manual
         Scenario: Create Secret: GO-01-TC02
-            Given user has imported gitops repo
+            # "user has imported gitops repo" looks too generic.
+            # There are several sub steps covered in - 
+            # https://github.com/redhat-developer/kam/tree/master/docs/journey/day1
+            Given user has imported gitops repo
```
Comment (amitkrout) [inline frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature:25]: Do you want me to expand it ?
ping @gajanan-more @sanketpathak

**Comment by amitkrout** [comment]
As per the screenshoot [1] expected in epic https://issues.redhat.com/browse/ODC-5157, I have added/updated the test scenario specific to environment page changes in developer console. However i do not see the same change on the latest (6th June) OCP 4.8 nightly build 

[1] 
Screen Shot 2021-06-07 at 3 02 01 PM

@reginapizza Would you please have a look.

**Comment by reginapizza** [comment]
@amitkrout you are seeing that view because the newest version of the gitops operator isn't be used I think? Only the newer version of the gitops operator will pass along the information for each environment's sync status and the lastest deployment time, otherwise it defaults to just a list of the dev environments and no time. Hope that clears up some confusion?

**Comment by amitkrout** [comment]
> @amitkrout you are seeing that view because the newest version of the gitops operator isn't be used I think? Only the newer version of the gitops operator will pass along the information for each environment's sync status and the lastest deployment time, otherwise it defaults to just a list of the dev environments and no time. Hope that clears up some confusion?

@reginapizza Your guess is correct. I tested the secnarion on GitOps Version 1.1.1. I think the new console changes will be reflected in GitOps 1.1.2 patch release happening shortly.

@baijum We can merge this changes

**Comment by baijum** [comment]
> @baijum We can merge this changes

@amitkrout ?!

File: frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature [hunk 2]
```diff
@@ -2,18 +2,18 @@
...
-
         Background:
-            Given user has installed gitOps operator
-              And user is at developer perspective
-
+            Given user has installed GitOps operator
+              And user has installed Pipelines operator
+              And user has installed Sealed Secrets operator in "cicd" namespace
+              And user has created Sealed Secrets instance of name "sealedsecretcontroller"
+              Then user is at developer perspective
```
Comment (makambalaji) [inline frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature:10]:
```suggestion
              And user is at developer perspective
```
In Background section, Only Given statements should be added

**Review by makambalaji — CHANGES_REQUESTED** [review]
One small modification needed

**Comment by amitkrout** [comment]
> > @baijum We can merge this changes
> 
> @amitkrout ?!

Ahh... i am sorry @baijum. I suppose to add @makambalaji instead. 

@baijum Please ignore

**Comment by amitkrout** [comment]
@makambalaji Addressed your review comment. Can you please check and merge if all look good.

File: frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature (hunk 2 above)
Comment (amitkrout) [reply frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature]: Done

**Comment by amitkrout** [comment]
> @makambalaji Addressed your review comment. Can you please check and merge if all look good.

Also looking for developer (@reginapizza) review. @reginapizza Please apply ```/lgtm``` in the comment if the test scenario looks good.

File: frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature [hunk 3]
```diff
@@ -22,17 +22,21 @@ Feature: Applications managed by GitOps and Create secret
...
+              And user can see the Applications on the page
+             Then user can add new environments
 
         @regression @manual
         Scenario: Application Details page for Applications: GO-01-TC03
             Given user is on the Environments page
               And user can see the Applications on the page
-             When user clicks on the application
-             Then user can see the application Details page
+             Then user can see the columns "Application Name", "Git repository", "Environments" and "Last deployment"
```
Comment (reginapizza) [inline frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature:35]: Based on their version of the GitOps operator, they will either see "Environments" (older; where all environments are in a comma-separated list) or "Environment Status" (newer; will have the # of environments for each status condition- Synced, OutOfSync, and Unknown)

Also, Application Name --> Application name (sentence case)

File: frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature [hunk 4]
```diff
@@ -22,17 +22,21 @@ Feature: Applications managed by GitOps and Create secret
               And user navigates to Environments page
              Then user will see the list GitOps application groupings on the page
 
+        @regression @manual
+        Scenario: Application Details page for Applications: GO-01-TC04
+            Given user is on the Environments page
+              And user can see the Applications on the page
+             Then user can add new environments
```
Comment (reginapizza) [inline frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature:29]: User cannot add new environments directly through console, if that's what this is implying

File: frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature (hunk 3 above)
Comment (amitkrout) [reply frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature:35]: @reginapizza Can you please share the support matrix. 

| GitOps Version  | OCP Version | Env Page  |
| ------------- | ------------- |  ------------- |
|   |   |   |
|   |   |   |

Comment (amitkrout) [reply frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature:35]: If GitOps version has the dependency then i think it will be easy to handle in the test implementation code. May be i can rephrase the step as 

```Then user can see the columns attributes depending upon the "GioOps" version```

File: frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature (hunk 4 above)
Comment (amitkrout) [reply frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature:29]: Yes, through its not possible. But using KAM user can add that. Let me rephrase it

Comment (amitkrout) [reply frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature:29]:
```
hen user can add new environments using kam cli
```
I think it makes more sense

**Comment by amitkrout** [comment]
@reginapizza Thanks for the review :) 

I have address your review comments. Can you please go for another round of review.

File: frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature (hunk 4 above)
Comment (reginapizza) [reply frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature]: perfect, thank you!

File: frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature (hunk 3 above)
Comment (reginapizza) [reply frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature]: working on confirming support matrix and I'll send that to you when I have it, but otherwise I think this statement is okay since it's generic.

File: frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature [hunk 5]
```diff
@@ -22,17 +22,21 @@ Feature: Applications managed by GitOps and Create secret
...
+              And user can see the Applications on the page
+             Then user can add new environments using kam cli
 
         @regression @manual
         Scenario: Application Details page for Applications: GO-01-TC03
             Given user is on the Environments page
               And user can see the Applications on the page
-             When user clicks on the application
-             Then user can see the application Details page
+             Then user can see the columns attributes depending upon the GioOps "version"
```
Comment (reginapizza) [inline frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature:35]: sorry, just one thing, typo here on GitOps

**Review by reginapizza — CHANGES_REQUESTED** [review]

File: frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature (hunk 5 above)
Comment (amitkrout) [reply frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature]: Done.
Thanks :)

**Comment by amitkrout** [comment]
> /lgtm

@gajanan-more Please reapply it.
